### PR TITLE
Adding the doctest-remote-data directive

### PR DESCRIPTION
**Pull request by bsipocz**
This is to fix the issue https://github.com/astropy/astroquery/issues/2111 and https://github.com/astropy/pytest-doctestplus/issues/161

The solution has been tested locally for astroquery where the code blocks now show up as they should.

### CODE REVIEW

**Comment by bsipocz** [comment]
After merging this we need to tag a new `sphinx-astropy` release...

**Comment by saimn** [comment]
The Sphinx extension was moved to pytest-doctestplus (we should deprecate it here, if not already done):
https://github.com/astropy/pytest-doctestplus/pull/113
https://github.com/astropy/pytest-doctestplus/tree/main/pytest_doctestplus/sphinx

**Comment by bsipocz** [comment]
🤦‍♀️ I'm sorry for the noise, I missed that.

**Comment by bsipocz** [comment]
Hmm, I'm not sure it ever made work to pick it up properly from pytest-doctestplus. I have to give it a more detailed look later.

However, given that this hasn't been properly deprecated here I would advocate for having the fix merged into both places.

**Comment by pllim** [comment]
Maybe Sphinx is confused when it exists in multiple places and picks up this one first? What if you completely remove it here; will Sphinx then picks up the one in doctestplus? 🤔

**Comment by bsipocz** [comment]
Yes, I'll do those experiments when scipy wraps up :)

**Comment by saimn** [comment]
Right indeed, we are still using `sphinx_astropy.ext.doctest` in the default sphinx config file, so we would need a new sphinx-astropy release deprecating `sphinx_astropy.ext.doctest` in favor of pytest-doctestplus and updating the defaut config.

**Comment by bsipocz** [comment]
OK, so this now relies of pytest-doctestplus to provide the `doctest-remote-data` directive. I've remove the previous patch with the assumption that pytest-doctestplus is required to make the directive work in the first place, so we can just as rely on it to provide the directives for the sphinx extension as well.

I'm not sure how to test it in the CI setting as it requires a new pytest-doctestplus release first, but locally it all work fine.

File: sphinx_astropy/conf/v1.py [hunk 1]
```diff
@@ -210,12 +210,21 @@ def check_sphinx_version(expected_version):
...
     'sphinx_astropy.ext.generate_config',
     'sphinx_astropy.ext.missing_static',
     'sphinx.ext.mathjax']
 
+try:
+    # Do this while in deprecation
+    import pytest_doctestplus.sphinx.doctestplus
+    extensions += ['pytest_doctestplus.sphinx.doctestplus']
+except ImportError:
+    warnings.warn("sphinx_astropy.ext.doctest has been deprecated "
```
Comment (pllim) [inline sphinx_astropy/conf/v1.py:223]:
```suggestion
    warnings.warn("sphinx_astropy.ext.doctest has been deprecated, "
```

File: sphinx_astropy/conf/v1.py [hunk 2]
```diff
@@ -210,12 +210,21 @@ def check_sphinx_version(expected_version):
...
+                  "use the doctestplus extensions from pytest-doctestplus: "
+                  "pytest_doctestplus.sphinx.doctestplus.",DeprecationWarning)
```
Comment (pllim) [inline sphinx_astropy/conf/v1.py:225]:
```suggestion
                  "pytest_doctestplus.sphinx.doctestplus", DeprecationWarning)
```

**Review by pllim — APPROVED** [review]
LGTM! After a release with this, we need to start removing the deprecated code? 😉

**Comment by pllim** [comment]
Hmm. Should we add `pytest-doctestplus` (with the new version about to be released), in https://github.com/astropy/sphinx-astropy/blob/main/setup.cfg ?

**Comment by bsipocz** [comment]
> Hmm. Should we add pytest-doctestplus (with the new version about to be released), in https://github.com/astropy/sphinx-astropy/blob/main/setup.cfg ?

Good question. I think we should, after all `sphinx-astropy` is an umbrella package.

> LGTM! After a release with this, we need to start removing the deprecated code? 😉

not sure how quickly we can do that, given a lot of packages may rely on sphinx-astropy and we don't want to create fire looking things that would make people panic and ping us... OTOH adding doctestplus properly as a dependency in setup.cfg, ideally no one would notice that anything has changed.

**Comment by saimn** [comment]
If we had a requirement on `pytest_doctestplus` then I guess there is no need for a deprecation, 'we can just remove the old extensions as the `pytest_doctestplus` one will be used instead.

**Comment by bsipocz** [comment]
I'm still worried a bit if anyone is using `sphinx_astropy.ext.doctest` directly rather than importing the `extensions` defined in the config file here.

File: sphinx_astropy/conf/v1.py [hunk 3]
```diff
@@ -18,6 +18,8 @@
 import sphinx
 from distutils.version import LooseVersion
 
+import pytest_doctestplus.sphinx.doctestplus
+
```
Comment (saimn) [inline sphinx_astropy/conf/v1.py:22]: Do we need the import ?

File: setup.cfg [hunk 4]
```diff
@@ -25,6 +25,7 @@ install_requires =
     sphinx-automodapi
     sphinx-gallery
     pillow
+    pytest-doctestplus
```
Comment (pllim) [inline setup.cfg:28]: Need a minversion pin here, right?

File: sphinx_astropy/conf/v1.py (hunk 3 above)
Comment (bsipocz) [reply sphinx_astropy/conf/v1.py:22]: Maybe we don't, I'm checking it now.

File: setup.cfg (hunk 4 above)
Comment (bsipocz) [reply setup.cfg:28]: I can, though most things should work with 0.9.0, too when the sphinx extension got moved here. Shall we have that or 0.10.1?

Comment (pllim) [reply setup.cfg:28]: Hmm... I would say 0.10.1 would be better since it fixes the code snippets, but is that too radical?

Comment (bsipocz) [reply setup.cfg:28]: I'm happy either way. I would say it's unlikely that anyone can pull this new version but has to stick with the previous from this January for the plugin.

**Comment by saimn** [comment]
Looks all good now ?

**Comment by pllim** [comment]
I'll take that as approval, @saimn . Merging!

**Comment by bsipocz** [comment]
Shall I go ahead with a new release then?

**Comment by saimn** [comment]
Yes, please go ahead :+1:

**Pull request merged**